eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -63,7 +64,10 @@ def lsqfitter(lc, model, meta, calling_function='lsq', **kwargs):
     model.update(fit_params, freenames)
 
     # Save the covariance matrix in case it's needed to estimate step size for a sampler
-    model_lc = model.eval()
+    if lc.share:
```

Comment:
I've tried updating the model.__init__ routine so that it loads in the parameters model.share, model.longparamlist, and model.nchan so that you can get rid of these extra parameters in model.eval!